api/src/pcapi/scripts/transform_virtual_venues/main.py

```diff
@@ -0,0 +1,248 @@
+                diffusible=False,
+                name="[ND]",
+                address=self.nd_address,
+                ape_code=ape_code,
+                ape_label=ape_label,
+                legal_category_code=self._legal_category_code(siret),
+            )
+
+        return SiretInfo(
+            siret=siret,
+            active=self._is_active(siret),
+            diffusible=True,
+            name="MINISTERE DE LA CULTURE",
+            address=self.address,
+            ape_code=ape_code,
+            ape_label=ape_label,
+            legal_category_code=self._legal_category_code(siret),
+        )
+
+
+def get_backend() -> EntrepriseWithHeadQuartersBackend | EntrepriseWithHeadQuartersTestingBackend:
+    if settings.IS_PROD or settings.IS_STAGING:
+        return EntrepriseWithHeadQuartersBackend()
+    return EntrepriseWithHeadQuartersTestingBackend()
+
+
+def venues_from_offerers_with_one_unique_virtual_venue() -> BaseQuery:
+    ids = {row[0] for row in db.session.execute(QUERY)}
+    return Venue.query.filter(Venue.id.in_(ids)).options(sa_orm.joinedload(Venue.managingOfferer))
+
+
+def get_siret(venue: Venue) -> SiretInfo:
+    """Find a venue's siret using its managing offerer's head quarters"""
+    if not venue.managingOfferer.siren:
+        raise MissingSirenError(f"venue #{venue.id}")
+
+    siren = venue.managingOfferer.siren[:9]
+    try:
+        siret_info = get_backend().get_head_quarter(siren)
+    except Exception as err:
+        raise FetchSiretError(f"venue #{venue.id} / siren: #{siren}") from err
+
+    return siret_info
+
+
+def get_address(venue: Venue, siret_info: SiretInfo) -> Address:
+    """Search address based on siret_info's geographical data.
+
+    If nothing is found, use the BAN API and create the missing address.
+    """
+    addresses = search_addresses(
+        street=siret_info.address.street, city=siret_info.address.city, postal_code=siret_info.address.postal_code
+    )
+
+    if addresses:
+        return addresses[0]
+
+    ban_address = _get_ban_address_or_none(
+        street=siret_info.address.street, postal_code=siret_info.address.postal_code, city=siret_info.address.city
+    )
+
+    if ban_address:
+        location_data = LocationData(
+            street=ban_address.street,
+            city=ban_address.city,
+            postal_code=ban_address.postcode,
+            insee_code=ban_address.citycode,
+            latitude=ban_address.latitude,
+            longitude=ban_address.longitude,
+            ban_id=ban_address.id,
+        )
+    else:
+        msg = (
+            f"venue #{venue.id} / "
+            f"street: '#{siret_info.address.street}' / "
+            f"city: '#{siret_info.address.city}' / "
+            f"postal code: '#{siret_info.address.postal_code}'"
+        )
+        raise FetchAddressError(msg)
+
+    return get_or_create_address(
+        location_data=location_data,
+        is_manual_edition=ban_address is None,
```

Comment:
D'après ton code précédent cette ligne sera toujours égal à `True`.
```suggestion
        is_manual_edition=True,
```